### PR TITLE
[GEOS-7463] GeoServer allows for layer specific catalog/services, but not for layer group ones

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/ows/LocalWorkspaceCallback.java [hunk 1]
```diff
@@ -57,14 +60,20 @@ public Request init(Request request) {
...
-                        }
-                        else {
-                            //TODO: perhaps throw an exception?
+                            LocalPublished.set(l);
+                        } else {
+                            lg = catalog.getLayerGroupByName(ws, last);
+                            if(lg != null) {
+                                LocalPublished.set(lg);
+                            } else {
+                                // TODO: perhaps throw an exception?
```
Comment (smithkm) [inline src/main/src/main/java/org/geoserver/ows/LocalWorkspaceCallback.java]: Probably worth a log warning at least.

**Comment by smithkm** [comment]
Looks good to me.

**Comment by aaime** [comment]
So do we want to merge it, or we consider it unfit for the feature freeze? This is in a gray area between a fix and a new feature.

**Comment by simboss** [comment]
I am going to merge this now on 2.10 and then discuss about eventual backport...

**Pull request merged**